### PR TITLE
Move sort_pairs() to libcrmcommon

### PR DESCRIPTION
**Pull request by kgaillot**
This is part of a larger project to move the functionality of libpe_rules into libcrmcommon and bring the code up to current guidelines.

### CODE REVIEW

**Comment by kgaillot** [comment]
@wenningerk , can you review this when you get a chance? Thanks

**Comment by kgaillot** [comment]
retest this please

File: lib/common/nvpair.c [hunk 1]
```diff
@@ -463,14 +479,15 @@ pcmk__cmp_nvpair_blocks(gconstpointer a, gconstpointer b, gpointer user_data)
...
-        return b_is_higher;
+    // NULL values have lowest priority
+    if (a == NULL) {
+        return (b == NULL)? 0 : b_is_higher;
+
     } else if (b == NULL) {
         return a_is_higher;
     }
 
+    // A particular XML ID can be specified as having highest priority
```
Comment (wenningerk) [inline lib/common/nvpair.c:491]: // XML ID is unique with all the blocks so we can't  have them both equal first ID
// and checking scores isn't needed here either

File: lib/common/nvpair.c [hunk 2]
```diff
@@ -463,14 +479,15 @@ pcmk__cmp_nvpair_blocks(gconstpointer a, gconstpointer b, gpointer user_data)
     const gint a_is_higher = unpack_data->overwrite? 1 : -1;
     const gint b_is_higher = -a_is_higher;
 
-    if (a == NULL && b == NULL) {
-        return 0;
-    } else if (a == NULL) {
-        return b_is_higher;
+    // NULL values have lowest priority
```
Comment (wenningerk) [inline lib/common/nvpair.c:483]: // no reason to turn that around for negative scores of the other that is != NULL

**Review by wenningerk — APPROVED** [review]
my comments are more questions - but having something like them might make reading the code easier

**Comment by wenningerk** [comment]
looking good

File: lib/common/nvpair.c (hunk 1 above)
Comment (kgaillot) [reply lib/common/nvpair.c:491]: Thanks, I'll update the comment

**Comment by kgaillot** [comment]
updated comments per review

**Pull request merged**